backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAO.java

```diff
@@ -62,5 +64,20 @@ BranchEntity getBranchEntity(Session session, Long repoId, String branchName)
   ListCommitsLogRequest.Response listCommitsLog(ListCommitsLogRequest request)
       throws ModelDBException;
 
-  FindRepositories.Response findRepositories(FindRepositories request) throws ModelDBException;
+  FindRepositories.Response findRepositories(FindRepositories request)
+      throws ModelDBException, InvalidProtocolBufferException;
+
+  void addRepositoryAttributes(RepositoryIdentification repositoryId, List<KeyValue> attributesList)
+      throws InvalidProtocolBufferException, ModelDBException;
+
+  void updateRepositoryAttribute(RepositoryIdentification repositoryId, KeyValue attribute)
+      throws InvalidProtocolBufferException, ModelDBException;
+
+  List<KeyValue> getRepositoryAttributes(
+      RepositoryIdentification repositoryId, List<String> attributeKeysList, boolean getAll)
+      throws InvalidProtocolBufferException, ModelDBException;
```

Comment:
we do not need this

Reply:
I have replaced `getDatasetAttributes` endpoint with `getRepositoryAttributes` that's why this is added.

Reply:
Remove `getRepositoryAttributes` endpoint